molviewspec/molviewspec/nodes.py

```diff
@@ -769,3 +769,182 @@ def validate_state_tree(json: str) -> None:
     """
     State.parse_raw(json)
+
+# TODO: fields instead of plain types
+class CircleParams(BaseModel):
+    kind: Literal["circle"] = "circle"
+    center: PrimitivePositionT = Field(description="The center of the circle.")
+    # TODO: elaborate names and semantics depending on
+    # how Mol* implements circles
+    # (should be dir_major and dir_minor perhaps as these two here are just Vec3)
+    major_axis: PrimitivePositionT = Field(description="Major axis of this circle.")
+    minor_axis: PrimitivePositionT = Field(description="Minor axis of this circle.")
+
+# TODO: add collection of descriptions for the fields with the same name
+class Polygon(BaseModel):
+    kind: Literal["polygon"] = "polygon"
+    vertices: list[float] = Field(description="3N length array of floats with vertex position (x1, y1, z1, ...)")
+    
+class Star(BaseModel):
+    kind: Literal["star"] = "star"
+    center: PrimitivePositionT = Field(description="The center of the star.")
+    inner_radius: float = Field(description="The inner radius of the star")
+    outer_radius: float = Field(description="The outer radius of the star")
+    # TODO: is this correct meaning?
+    point_count: int = Field(description="The number of points the star contains")
+    # TODO: inherit from TransformParams instead?
+    # rotation: Optional[Mat3[float]] = Field(
+    #     description="9d vector describing the rotation, in a column major (j * 3 + i indexing) format, this is equivalent to Fortran-order in numpy, to be multiplied from the left",
+    # )
+
+class TransformMixin(BaseModel):
+    # rotation: Optional[Mat3[float]] = Field(
+    #     description="9d vector describing the rotation, in a column major (j * 3 + i indexing) format, this is equivalent to Fortran-order in numpy, to be multiplied from the left",
+    # )
+    rotation_axis: Optional[Vec3] = Field(
+        # TODO: better description
+        description="Rotation axis"
+    )
+    rotation_radians: Optional[int] = Field(
+        description="Rotation angle in radians"
+    )
+    translation: Optional[Vec3[float]] = Field(description="3d vector describing the translation")
+
+class RectangularPrismMixin(BaseModel):
+    center: PrimitivePositionT = Field(description="The center of the box.")
+    extent: Vec3 = Field(description="The width, the height, and the depth of the box.")
+    # TODO: include in TransformParams instead?
+    scaling: Optional[Vec3[float]] = Field(description="3d vector describing the scaling.")
+    alignment: Literal["axes", "principal-axes"] = "axes"
+    groups: Optional[list[int]] = Field(description="Assign a number to each box to group them.")
+    color: Optional[ColorT] = Field(description="Default color for the box.")
+    
+class BoxParams(TransformMixin, RectangularPrismMixin):
+    kind: Literal["box"] = "box"
+    
+class CageParams(TransformMixin, RectangularPrismMixin):
+    kind: Literal["cage"] = "cage"
+    type: Literal["as_lines", "as_geometry"] = "as_lines"
```

Comment:
```suggestion
    render_style: Literal["lines", "geometry"] = "lines"
```

Or something like that.